ddtrace/appsec/processor.py

```diff
@@ -68,36 +69,47 @@ class _Addresses(object):
     SERVER_RESPONSE_HEADERS_NO_COOKIES = "server.response.headers.no_cookies"
 
 
-_COLLECTED_REQUEST_HEADERS = {
-    "accept",
-    "accept-encoding",
-    "accept-language",
-    "content-encoding",
-    "content-language",
-    "content-length",
-    "content-type",
-    "forwarded",
-    "forwarded-for",
-    "host",
-    "true-client-ip",
-    "user-agent",
-    "via",
-    "x-client-ip",
-    "x-cluster-client-ip",
-    "x-forwarded",
-    "x-forwarded-for",
-    "x-real-ip",
-}
-
-_COLLECTED_HEADER_PREFIX = "http.request.headers."
-
-
-def _set_headers(span, headers):
-    # type: (Span, Dict[str, Union[str, List[str]]]) -> None
+_COLLECTED_REQUEST_HEADERS = frozenset(
+    {
+        "accept",
+        "accept-encoding",
+        "accept-language",
+        "content-encoding",
+        "content-language",
+        "content-length",
+        "content-type",
+        "forwarded",
+        "forwarded-for",
+        "host",
+        "true-client-ip",
+        "user-agent",
+        "via",
+        "x-client-ip",
+        "x-cluster-client-ip",
+        "x-forwarded",
+        "x-forwarded-for",
+        "x-real-ip",
+    }
+)
+
+
+_ALWAYS_COLLECT_HEADERS = frozenset(
+    {
+        "content-length",
+        "content-type",
+        "content-encoding",
+        "content-language",
+    }
+)
+
+
+def _set_headers(span, kind, headers, extra_collect=None):
+    # type: (Span, str, Dict[str, Union[str, List[str]]], Optional[frozenset[str]]) -> None
     for k in headers:
-        if k.lower() in _COLLECTED_REQUEST_HEADERS:
+        low = k.lower()
+        if low in _ALWAYS_COLLECT_HEADERS or low in (extra_collect or {}):
```

Comment:
can we do `extra_collect or {}` outside the loop, e.g.
~~~ python
extra_collect = extra_collect or {}
~~~
just before the `for`?